sageresearch-app-sdk/src/main/java/org/sagebionetworks/research/sageresearch_app_sdk/TaskResultUploader.java

```diff
@@ -103,65 +102,85 @@ public TaskResultUploader(@NonNull final BridgeConfig bridgeConfig,
     public Completable processTaskResult(final TaskResult taskResult) {
         LOGGER.info("Uploading task result: {}", taskResult);
 
-        SchemaKey sk = bridgeConfig.getTaskToSchemaMap().get(taskResult.getIdentifier());
+        SchemaKey schemaKey = bridgeConfig.getTaskToSchemaMap().get(taskResult.getIdentifier());
 
-        if (sk == null) {
-            LOGGER.warn("No schema found for task " + taskResult.getIdentifier() +  ". Revision 1 will be used.");
-            sk = new SchemaKey(taskResult.getIdentifier(), 1);
+        if (schemaKey == null) {
+            LOGGER.warn("No schema found for task " + taskResult.getIdentifier() + ". Revision 1 will be used.");
+            schemaKey = new SchemaKey(taskResult.getIdentifier(), 1);
         }
 
-        Archive.Builder builder = Archive.Builder.forActivity(sk.getId(), sk.getRevision());
-        String appVersionString = String.format(Locale.ENGLISH, "version %s, build %d",
-                bridgeConfig.getAppVersionName(),
-                bridgeConfig.getAppVersion());
-
-        builder.withAppVersionName(appVersionString)
-                .withPhoneInfo(bridgeConfig.getDeviceName());
-
-        for (ArchiveFile resultArchiveFile : abstractResultArchiveFactory.toArchiveFiles(taskResult)) {
-            builder.addDataFile(resultArchiveFile);
-        }
+        String archiveFilename = schemaKey.getId() + schemaKey.getRevision() + taskResult.getTaskUUID();
 
-        String archiveFilename = sk.getId() + sk.getRevision() + taskResult.getTaskUUID();
-
-        Completable uploadToS3Completable = Completable.fromAction(() ->
-                uploadManager.queueUpload(archiveFilename, builder.build())

```

Comment:
Hard to see bug. The `Action` gets called but the `rx.Completable` it produces is ignored (never attached to the rest of the Rx chain).

Reply:
I don't understand, why wouldn't this code... 

```
return metadataCompletable(builder, taskResult)
                 .concatWith(uploadToS3Completable);
```

Not concatenate the upload to S3 action -> flatMapCompletable (upload it to s3).  What am I missing about RX here?